eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.test.js

```diff
@@ -60,51 +61,109 @@ describe("Row", () => {
-        {
-          questionnaire,
-          onDeleteQuestionnaire: handleDeleteQuestionnaire,
-          onDuplicateQuestionnaire: handleDuplicateQuestionnaire,
-          foo: "bar",
-        },
-        { showDeleteQuestionnaireDialog: false }
-      )
-    ).toBeFalsy();
+  it("should call push when link is clicked", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    const tableCell = wrapper.find(TD).first();
+    tableCell.simulate("click");
+    expect(props.history.push).toHaveBeenCalled();
+  });
+
+  it("should only grab focus once", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+
+    const focus = jest.fn();
+    const getElementsByTagName = jest.fn(() => [{ focus }]);
+    const instance = wrapper.instance();
+    instance.rowRef = { current: { getElementsByTagName } };
+
+    wrapper.setProps({
+      autoFocus: true,
+    });
+
+    wrapper.setProps({
+      autoFocus: true,
+    });
+
+    expect(focus).toHaveBeenCalledTimes(1);
+  });
+
+  it("should allow duplication of a Questionnaire", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+
+    wrapper.find(DuplicateQuestionnaireButton).simulate("click");
+
+    expect(props.onDuplicateQuestionnaire).toHaveBeenCalledWith(questionnaire);
   });
 
   it("should call delete confirm dialog when the delete button is clicked", () => {
-    const wrapper = shallow(
-      <Row
-        questionnaire={questionnaire}
-        onDeleteQuestionnaire={handleDeleteQuestionnaire}
-        onDuplicateQuestionnaire={handleDuplicateQuestionnaire}
-      />
-    );
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
     wrapper.find(IconButtonDelete).simulate("click");
 
-    expect(handleDeleteQuestionnaire).toHaveBeenCalledWith(questionnaire);
+    expect(props.onDeleteQuestionnaire).toHaveBeenCalledWith(questionnaire);
+  });
+
+  it("should show the short title when it is provided", () => {
+    let wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    let shortTitle = wrapper.find(ShortTitle);
+    expect(shortTitle).toMatchSnapshot();
+
+    const originalShortTitle = questionnaire.shortTitle;
+    questionnaire.shortTitle = "";
+
+    wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    expect(wrapper.find(ShortTitle)).toHaveLength(0);
+
+    questionnaire.shortTitle = originalShortTitle;
+  });
+
+  it("should call focusLink in componentDidMount when autoFocus is true", () => {
+    const focusLink = jest.fn();
+    const originalFunc = Row.prototype.focusLink;
+    Row.prototype.focusLink = focusLink;
+    shallow(<Row questionnaire={questionnaire} autoFocus {...props} />);
+    expect(focusLink).toHaveBeenCalled();
+    Row.prototype.focusLink = originalFunc;
+  });
+
+  it("should style table row correctly", () => {
+    const tableRow = shallow(<TR />);
```

Comment:
You could pass directly instead of doing this in two steps.

Reply:
But we can completely get rid of the need for this test if do this in pure css.

Reply:
Removed this test as I am using pure css instead.